client/lib/mixins/analytics/index.js

```diff
@@ -81,6 +81,56 @@ const EVENTS = {
 				'calypso_domain_search',
 				{
 					search_box_value: searchBoxValue,
+					time_diff_from_last_search: timeDiffFromLastSearch,
+					search_count: searchCount,
+					section
+				}
+			);
+		},
+
+		searchFormView( section ) {
+			analytics.ga.recordEvent(
+				'Domain Search',
+				'Landed on Search'
+			);
+
+			analytics.tracks.recordEvent( 'calypso_domain_search_pageview', { section } );
+		},
+
+		searchResultsReceive( searchQuery, searchResults, responseTimeInMs, resultCount, section ) {
+			analytics.ga.recordEvent(
+				'Domain Search',
+				'Receive Results',
+				'Response Time',
+				responseTimeInMs
+			);
+
+			analytics.tracks.recordEvent(
+				'calypso_domain_search_results_suggestions_receive',
+				{
+					search_query: searchQuery,
+					results: searchResults.map( result => result.domain_name ).join( ';' ),
```

Comment:
If you pass it an error from the backend (as you do [above](https://github.com/Automattic/wp-calypso/pull/5100/files#diff-99f1f1d8ba147cd06cfd4a0d8d195d34R322)), it will fail, as there's, of course, no `domain_name` property on that.